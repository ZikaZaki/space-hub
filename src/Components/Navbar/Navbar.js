import * as React from 'react';
import { NavLink } from 'react-router-dom';
import styles from './Navbar.module.css';
// import the nav-logo image
import NavLogo from './planet.png';

const Navbar = () => (
  <div className={styles['nav-container']}>
    <nav className={styles.nav}>
      <div className={styles['nav-logo']}>
        <img className={styles['logo-img']} src={NavLogo} alt="planet" />
        <h1 className={styles['logo-title']}>Space Travelers&apos; Hub</h1>
      </div>
      <ul className={styles['nav-list']}>
        <li className={styles['list-item']}>
          <NavLink
            className={({ isActive }) => (isActive ? styles['active-link'] : styles['inactive-link'])}
            end
            to="/"
          >
            Rockets
          </NavLink>
        </li>
        <li className={styles['list-item']}>
          <NavLink
            className={({ isActive }) => (isActive ? styles['active-link'] : styles['inactive-link'])}
            to="/dragons"
          >
            Dragons
          </NavLink>
        </li>
        <li className={styles['list-item']}>
          <NavLink
            className={({ isActive }) => (isActive ? styles['active-link'] : styles['inactive-link'])}
            to="/missions"
          >
            Missions
          </NavLink>
        </li>
        <span className={styles.separator}> | </span>
        <li className={styles['list-item']}>
          <NavLink
            className={({ isActive }) => (isActive ? styles['active-link'] : styles['inactive-link'])}
            to="/profile"
          >
            My Profile
          </NavLink>
        </li>
      </ul>
    </nav>
    <hr />
  </div>
);

export default Navbar;
