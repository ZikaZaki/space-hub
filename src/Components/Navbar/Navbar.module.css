.nav-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  background-color: #fff;
  height: 8rem;
  width: 100%;
  padding: 1rem 2rem;
  margin: 0;
}

.nav-container hr {
  font-weight: bold;
  color: #121212;
  background-color: #121212;
  height: 1.5px;
  width: 100%;
  padding: 0;
  margin: 0;
}

.nav {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  background-color: #fff;
  height: 7rem;
  width: 100%;
  padding: 1rem;
  margin: 0;
}

.nav-logo {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
  gap: 1rem;
  width: 100%;
  height: 100%;
  padding: 0 1rem;
  margin: 0;
}

.logo-img {
  height: 80px;
  width: 80px;
  padding: 0;
  margin: 0;
}

.logo-title {
  display: flex;
  font-family: var(--montserrat);
  font-size: 1.875rem;
  font-weight: 800;
  font-weight: bold;
  color: black;
  text-align: justify;
  min-width: 16rem;
  margin: 0;
  padding: 0;
}

.nav-list {
  font-family: var(--montserrat);
  font-weight: bold;
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  align-items: center;
  text-align: center;
  gap: 1rem;
  color: #0290ff;
  list-style: none;
  width: 100%;
  padding: 0 1rem;
  margin: 0;
}

.list-item {
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.nav-link {
  font-size: 0.9rem;
  letter-spacing: 1.9px;
  text-decoration: none;
  padding: 0.2rem;
}

.separator {
  color: #121212;
}

.nav-link:active {
  text-decoration: underline;
}
