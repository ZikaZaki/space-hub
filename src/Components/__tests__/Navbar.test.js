import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Navbar from '../Navbar/Navbar';
import store from '../../redux/configureStore';
import '@testing-library/jest-dom';

it('Check if the component has changed', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Navbar />
      </Provider>
    </BrowserRouter>,

  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render the main container', () => {
  render(
    <BrowserRouter>
      <Provider store={store}>
        <Navbar />
      </Provider>
    </BrowserRouter>,
  );

  const container = screen.getByTestId('nav-container');

  expect(container).toBeInTheDocument();
});
