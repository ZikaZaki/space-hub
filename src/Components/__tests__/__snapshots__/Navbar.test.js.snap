// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if the component has changed 1`] = `
<div
  className="nav-container"
  data-testid="nav-container"
>
  <nav
    className="nav"
  >
    <div
      className="nav-logo"
    >
      <img
        alt="planet"
        className="logo-img"
        src="planet.png"
      />
      <h1
        className="logo-title"
      >
        Space Travelers' Hub
      </h1>
    </div>
    <ul
      className="nav-list"
    >
      <li
        className="list-item"
      >
        <a
          aria-current="page"
          className="active-link"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          className="inactive-link"
          href="/dragons"
          onClick={[Function]}
        >
          Dragons
        </a>
      </li>
      <li
        className="list-item"
      >
        <a
          className="inactive-link"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <span
        className="separator"
      >
         | 
      </span>
      <li
        className="list-item"
      >
        <a
          className="inactive-link"
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </nav>
  <hr />
</div>
`;
